### ISSUE TITLE
lunarvim hangs opening 200mb JSON doc

### ISSUE THREAD

**Issue by zapman449** [LunarVim/LunarVim#3393]
### Problem description

1. Open large JSON blob (my example is technically JSONL, but same for JSON)
2. lunarvim hangs

Example snippit:

```json
[
  "hbt-p-us-west-2",
  "a2luZDogc3EtY2xvdWQtY2QKc2
G5hbWU6IEhCVF9TS0lfSU1BR0UKIC
mZjCiAgICAgIHN0YWdpbmctdXMtd2
U1CREFfVkVSU0lPTgogICAgdGFyZ2
WU6IEFXU19BQ0NPVU5UX0lECiAgIC
3QtMgo="
]
[
  "hbt-p-us-east-1",
  "a2luZDogc3EtY2xvdWQtY2QKc2
G5hbWU6IEhCVF9TS0lfSU1BR0UKIC
GQyCiAgICAgIHN0YWdpbmctdXMtZW
U1CREFfVkVSU0lPTgogICAgdGFyZ2
iBBV1NfQUNDT1VOVF9JRAogICAgdG
TEK"
]
...
```

Note: for disclosure reasons, I cut off most of the base64 content (truncated each line of the line-wrapped string value), but it's nothing special

NOTE: neovim 0.8 opens the file fine.

FWIW:

```
❯ ls -lh *.json
-rw-r--r-- 1 jprice 259M Nov  1 17:06 o2.json
-rw-r--r-- 1 jprice 392M Nov  1 16:33 output.json
```

I can hang lvim with either of those, but o2.json is the simpler format indicated above.

### LunarVim version

master-e4a5fe9

### Neovim version (>= 0.8.0)

NVIM v0.8.0

### Operating system/version

macos 12.6

### Steps to reproduce

see above

### support info

```console
 Use [q] or [Esc] to quit the window

 Language client log: /Users/jprice/.local/state/nvim/lsp.log
 Detected filetype:   alpha

 0 client(s) attached to this buffer:

 Configured servers list:
```

```console
    __                          _    ___
   / /   __  ______  ____ _____| |  / (_)___ ___
  / /   / / / / __ \/ __ `/ ___/ | / / / __ `__ \
 / /___/ /_/ / / / / /_/ / /   | |/ / / / / / / /
/_____/\__,_/_/ /_/\__,_/_/    |___/_/_/ /_/ /_/


Buffer info
* filetype:                alpha
* bufnr:                   1
* treesitter status:       inactive

Active client(s)

Automatic LSP info
* Skipped servers: [angularls, ansiblels, csharp_ls, cssmodules_ls, denols, ember, emmet_ls, eslint, golangci_lint_ls, graphql, jedi_language_server, ltex, phpactor, psalm, pylsp, quick_lint_js, rome, reason_ls, solang,

Formatters info
* Active:
* Supported: []

Linters info
* Active:
* Supported: []

Code actions info
* Active:
```

**Comment by LostNeophyte** [comment]
#3268 probably fixed it. Update lunarvim

**Comment by zapman449** [comment]
Updating improved things... the file loads now.

however, it hangs after rendering the first page of the file now.  No input is accepted in command mode.  can't `:q` either.

**Comment by LostNeophyte** [comment]
> Updating improved things... the file loads now.
> 
> however, it hangs after rendering the first page of the file now. No input is accepted in command mode. can't `:q` either.

so you can't even edit the file?

**Comment by zapman449** [comment]
Correct, I can not edit the file.  (cant exit lvim either... have to kill -7 the PID)

**Comment by zapman449** [comment]
FWIW, my config is bog standard... only custom setting is `vim.opt.mouse = ""`

**Comment by abzcoding** [comment]
@zapman449 
can you try adding this to your config?
```lua
  vim.cmd [[
" disable syntax highlighting in big files
function! DisableSyntaxTreesitter()
    echo("Big file, disabling syntax, treesitter and folding")
    if exists(':TSBufDisable')
        exec 'TSBufDisable autotag'
        exec 'TSBufDisable highlight'
    endif

    set foldmethod=manual
    syntax clear
    syntax off
    filetype off
    set noundofile
    set noswapfile
    set noloadplugins
    set lazyredraw
endfunction

augroup BigFileDisable
    autocmd!
    autocmd BufReadPre,FileReadPre * if getfsize(expand("%")) > 1024 * 1024 | exec DisableSyntaxTreesitter() | endif
augroup END
  ]]
```

**Comment by LostNeophyte** [comment]
> @abzcoding You totally save my days. This works for me beautifully. But do you think it's possible to still enable TS somehow? Like, not enabling TS entirely.

you can try running `:TSBufEnable highlight` to see if won't be too slow, but it probably will

**Comment by abzcoding** [comment]
> @abzcoding You totally save my days. This works for me beautifully. But do you think it's possible to still enable TS somehow? Like, not enabling TS entirely.

you're welcome, I should probably rewrite this in lua, also not sure about treesitter performance, I need to test it

**Comment by LostNeophyte** [comment]
> > @abzcoding You totally save my days. This works for me beautifully. But do you think it's possible to still enable TS somehow? Like, not enabling TS entirely.
> 
> you're welcome, I should probably rewrite this in lua, also not sure about treesitter performance, I need to test it

it's here 978ff7c

**Comment by abzcoding** [comment]
> it's here [978ff7c](https://github.com/LunarVim/LunarVim/commit/978ff7c24de968d5f2f956b72a196773ae80b903) (mostly)

yes i've seen that before, sadly that doesn't help much with the performance, hence the need for this vim.cmd

**Comment by LostNeophyte** [comment]
yea I'm wondering why, it was based on your autocmd btw maybe you have treesitter foldexpr / intent blankline show_current_context enabled?

**Comment by abzcoding** [comment]
> yea I'm wondering why, it was based on your autocmd btw maybe you have treesitter foldexpr / intent blankline show_current_context enabled?

Yeah I should investigate it, maybe when the situation in my country becomes a bit more stable, I'll dig deeper into this

**Comment by LostNeophyte** [comment]
> > yea I'm wondering why, it was based on your autocmd btw maybe you have treesitter foldexpr / intent blankline show_current_context enabled?
> 
> Yeah I should investigate it, maybe when the situation in my country becomes a bit more stable, I'll dig deeper into this

If you have a big file you could share I could do that. I hope the situation gets stable soon :heart:

**Comment by abzcoding** [comment]
> If you have a big file you could share I could do that. I hope the situation gets stable soon ❤️

thanks, you can start with [this](https://github.com/json-iterator/test-data/blob/master/large-file.json)
and then [this](https://github.com/zemirco/sf-city-lots-json/blob/master/citylots.json)
